New installation handling

plerup opened the pull request:
A total rewrite of the installation concept has been made. The installation scripts now uses the nrf-connect toolchain manager to install a separate environment directory structure with all the required tools. This means that is no interference with possible already installed tools will occur.The installation will now use version 2.2.0 of nrf-connect.
For the Windows installation only the parts which require it will run with admin rights and hence keeping virus program happier.

Use of the correct environment is automatically handled via the "do" command.

The ubxlib submodule has been updated to use latest version.

New "do" command "terminal" which will launch a new shell instance where the correct environment is setup in case usage of the version specific tools.

plerup committed:
New installation handling

A total rewrite of the installation concept has been made.
The installation scripts now uses the nrf-connect toolchain manager
to install a separate environment directory structure with all the
required tools. This means that is no interference with possible
already installed tools will occur.The installation will now use
version 2.2.0 of nrf-connect.
For the Windows installation only the parts which require it will run
with admin rights and hence keeping virus program happier.

Use of the correct environment is automatically handled via the
"do" command.

The ubxlib submodule has been updated to use latest version.

New "do" command "terminal" which will launch a new shell instance
where the correct environment is setup in case usage of the version
specific tools.

philwareublox commented on install/install_windows.bat:
```diff
@@ -12,119 +12,156 @@ rem distributed under the License is distributed on an "AS IS" BASIS,
+  del %NEWT_TAR_FILE%
+)
+
+set DO_CMD=do.bat
```
There is no do.bat file anymore - it's just do.py

plerup replied:
They are now created by the installation and are no longer part of the repo

philwareublox commented on install/install_windows.bat:
```diff
@@ -12,119 +12,156 @@ rem distributed under the License is distributed on an "AS IS" BASIS,
+
+set DO_CMD=do.bat
+echo @set NCS_DIR=%NCS_DIR%>%DO_CMD%
+echo @set PATH=^%%PATH^%%;^%%ProgramFiles^%%\Microsoft VS Code\bin>>%DO_CMD%
```
I don't have a do.bat command at this location on my laptop.
![image](https://github.com/u-blox/ubxlib_examples_xplr_iot/assets/20516919/f2c0362f-9b75-462c-905f-6051d8222570)

RobMeades reviewed (approved):
Quite a rewrite!  Looks good as far as I can tell [which probably isn't very far :-)].

plerup committed:
Some cosmetic changes

Avoid VS Code extensions which can cause confusing messages
Removed some deprecated fields in the overlay

philwareublox reviewed (approved):
Sorry, I don't have the bandwidth to complete this review.
